com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Shaders/TemporalAntiAliasing.shader

```diff
@@ -141,45 +172,79 @@ Shader "Hidden/HDRP/TemporalAA"
 
     // ------------------------------------------------------------------
 
+        // This complexity will not be needed when moving to CS.
+        void SwizzleFilterWeights(int2 posSS, inout float4 filterParams1, inout float4 filterParams2)
+        {
+            // Data arrives as if filterParams weights for { (0, 1), (1, 0), (-1, 0), (0,-1) }, filterParams2 for { (-1, 1), (1, -1), (1, 1), (-1, -1) }
+            bool2 needSwizzle = (posSS & 1) == 0;
+
+            filterParams1.yz = needSwizzle.x ? filterParams1.zy : filterParams1.yz;
+            filterParams1.xw = needSwizzle.y ? filterParams1.wx : filterParams1.xw;
+#if WIDE_NEIGHBOURHOOD
+            filterParams2 = all(needSwizzle) ? filterParams2.yxwz :
+                (needSwizzle.x && !needSwizzle.y) ? filterParams2.zwxy :
+                (!needSwizzle.x && needSwizzle.y) ? filterParams2.wzyx : filterParams2;
+#endif
+        }
+
         void FragTAA(Varyings input, out CTYPE outColor : SV_Target0)
         {
             UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(input);
 
             float sharpenStrength = _TaaFrameInfo.x;
             float2 jitter = _TaaJitterStrength.zw;
 
-            float2 uv = input.texcoord - jitter;
+            float2 uv = input.texcoord;
+
+            #ifdef TAA_UPSCALE
+            float2 outputPixInInput = input.texcoord * _InputSize.xy - _TaaJitterStrength.xy;
+
+            uv = _InputSize.zw * (0.5f + floor(outputPixInInput));
+            #endif
 
             // --------------- Get closest motion vector ---------------
             float2 motionVector;
 
 #if ORTHOGRAPHIC
-            float2 closest = input.positionCS.xy;
+            float2 closestOffset = 0;
 #else
-            float2 closest = GetClosestFragment(_DepthTexture, int2(input.positionCS.xy));
+            float2 closestOffset = GetClosestFragmentOffset(_DepthTexture, int2(input.positionCS.xy));
 #endif
-            DecodeMotionVector(LOAD_TEXTURE2D_X(_CameraMotionVectorsTexture, closest), motionVector);
+
+            DecodeMotionVector(SAMPLE_TEXTURE2D_X_LOD(_CameraMotionVectorsTexture, s_point_clamp_sampler, ClampAndScaleUVForPoint(uv + closestOffset * _InputSize.zw), 0), motionVector);
             // --------------------------------------------------------
 
             // --------------- Get resampled history ---------------
             float2 prevUV = input.texcoord - motionVector;
 
-            CTYPE history = GetFilteredHistory(_InputHistoryTexture, prevUV, _HistorySharpening, _TaaHistorySize);
+            CTYPE history = GetFilteredHistory(_InputHistoryTexture, prevUV, _HistorySharpening, _TaaHistorySize, _RTHandleScaleForTAAHistory);
```

Comment:
Maybe add a small comment explaining why we need a separate scale here.

Reply:
Added on top where the scales are defined